src/domains/wallet-app/api/wallet-service/v1/_openapi.json.tpl

```diff
@@ -1,248 +1,368 @@
+        "required": [
+          "paymentInstrumentType",
+          "bin",
+          "maskedPan",
+          "expiryDate"
+        ]
+      },
+      "WalletCreateRequest": {
+        "type": "object",
+        "description": "Wallet creation request",
+        "properties": {
+          "paymentInstrumentType": {
+            "$ref": "#/components/schemas/PaymentInstrumentType"
+          },
+          "services": {
+            "type": "array",
+            "description": "List of services for which wallet is enabled",
+            "minItems": 1,
+            "items": {
+              "$ref": "#/components/schemas/Service"
             }
           }
         },
-        "WalletPostResponse": {
-          "type": "object",
-          "properties": {
-            "walletItemId": {
-              "type": "string"
-            },
-            "logoutURI": {
-              "type": "string"
-            }
+        "required": [
+          "paymentInstrumentType",
+          "services"
+        ]
+      },
+      "WalletCreateResponse": {
+        "type": "object",
+        "description": "Wallet creation response",
+        "properties": {
+          "walletId": {
+            "$ref": "#/components/schemas/WalletId"
+          },
+          "redirectUrl": {
+            "type": "string",
+            "format": "url",
+            "description": "Redirection URL to a payment gateway page where the user can input a payment instrument information",
+            "example": "http://localhost/inputPage"
           }
         },
-        "WalletUpdate": {
-          "type": "object"
-        },
-        "WalletUpdateResponse": {
-          "type": "object"
+        "required": [
+          "walletId",
+          "redirectUrl"
+        ]
+      },
+      "WalletInfo": {
+        "type": "object",
+        "description": "Wallet information",
+        "properties": {
+          "walletId": {
+            "$ref": "#/components/schemas/WalletId"
+          },
+          "userId": {
+            "type": "string",
+            "description": "user identifier"
+          },
+          "status": {
+            "$ref": "#/components/schemas/WalletStatus"
+          },
+          "creationDate": {
+            "description": "Wallet creation date",
+            "type": "string",
+            "format": "date-time"
+          },
+          "updateDate": {
+            "description": "Wallet update date",
+            "type": "string",
+            "format": "date-time"
+          },
+          "paymentInstrumentType": {
+            "$ref": "#/components/schemas/PaymentInstrumentType"
+          },
+          "paymentInstrumentId": {
+            "description": "Payment instrument identifier",
+            "type": "string"
+          },
+          "contractNumber": {
+            "description": "User contract identifier to be used with payment instrument to make a new payment",
+            "type": "string"
+          },
+          "gatewaySecurityToken": {
```

Comment:
better not to return this info to the client 😁

Reply:
fixed with https://github.com/pagopa/pagopa-infra/pull/903/commits/d16239dffa5356b30293b97b85c34b6b83d69df7